### PR TITLE
Remove shared folder

### PR DESCRIPTION
**Pull request by eldiegod**
fixes:  #1541

Removes shared folder, fixes components file structure inconsistencies and adds folders like forms, overalys, icons and illustrations into the components folder

### CODE REVIEW

**Comment by kdembler** [comment]
We have this entry in `.prettierignore`:
```
# don't prettify minified animation JSONs
src/shared/assets/animations
```
It needs to be updated because your PR prettifies all the animations causing this massive diff. They should be left as they are, minified

**Comment by eldiegod** [comment]
@kdembler  fixed

**Comment by drillprop** [comment]
My suggestions/feedback:
- I think we could move some components to views, especialy the ones which have name with suffix Studio/Viewer for example SidenavStudio, SidenavViewer
- we could group Buttons together. Keep Button, ButtonBase, IconButton in one directory
- Select should go to forms
- Maybe we could move TransactionModal also to views/studio?
- let's move TermsOfService to views/legal

**Comment by eldiegod** [comment]
> My suggestions/feedback:
> 
>     * I think we could move some components to views, especialy the ones which have name with suffix Studio/Viewer for example SidenavStudio, SidenavViewer
> 
>     * we could group Buttons together. Keep Button, ButtonBase, IconButton in one directory
> 
>     * Select should go to forms
> 
>     * Maybe we could move TransactionModal also to views/studio?
> 
>     * let's move TermsOfService to views/legal

- ButtonBase should just be under Button imo and we should get rid of IconButton idk why we maintain it when Button can have icon only prop
- Maybe rename `forms` to `inputs`
- TransactionModal is not to clear to me where it should go
- agree with the rest

**Comment by kdembler** [comment]
There's ongoing discussion in RocketChat on what the structure should be.

Other than that, my suggestions:
- `InputBase`, `OptionCard`, `Select` should be in `forms` (I would vote for also moving `FileSelect` and `MultiFileSelect` here) (possibly rename to `inputs`?)
- `Popover`, `ContextMenu`, `StepperModal`, `TransactionModal` should be in `overlays`
- Create `navigation` category that would contain all topbar/sidenav/bottomnav components
- Create `channel` category for all channel related stuff
- Create `video` category for all video related stuff
- Create `content` (possibly could use a better name) to keep all the content sections - `TopTenThisWeek`, `DiscoverChannels`, etc.

Reply to @drillprop feedback:
- I think `buttons` category works as well
- If possible I wouldn't want to move too many components to `views`/. `TransactionModal` will be soon used by viewer as well
- `TermsOfService` is used by studio as well, in sign in so needs to stay a generic component

**Comment by eldiegod** [comment]
Added the folders mentioned; Button is an input so figured I'd put it in `_inputs` along with select and also avoided moving components to views for now

**Comment by kdembler** [comment]
Personally, I'd move `Button` out of `_inputs`, doesn't feel like the right place for it for me. I suggest moving it to `_buttons` along with other button-like components that are currently uncategorized. This isn't a strong opinion though, if there's an agreement it should be in `_inputs`, I'm fine with that

File: src/components/Badge/Badge.styles.tsx [hunk 1]
```diff
@@ -1,9 +1,8 @@
 import { css } from '@emotion/react'
```
Comment (rafalpawlow) [inline src/components/Badge/Badge.styles.tsx]: Can we get rid of `.tsx` extension as part of this PR?

File: src/components/CircularProgress/CircularProgress.styles.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/components/CircularProgress/CircularProgress.styles.tsx]: Get rid of `.tsx` extension

File: src/components/FiltersBar/FiltersBar.styles.tsx [hunk 3]
```diff
@@ -1,9 +1,9 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/components/FiltersBar/FiltersBar.styles.tsx:1]: Get rid of `.tsx` extension

File: src/components/GlobalStyles/transitionStyles.tsx [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import { css } from '@emotion/react'
```
Comment (rafalpawlow) [inline src/components/GlobalStyles/transitionStyles.tsx]: Get rid of `.tsx` extension

File: src/components/Pagination/Pagination.styles.tsx [hunk 5]
```diff
@@ -1,14 +1,13 @@
 import { css } from '@emotion/react'
```
Comment (rafalpawlow) [inline src/components/Pagination/Pagination.styles.tsx]: Get rid of `.tsx` extension

File: src/components/ProgressDrawer/ProgressDrawer.styles.tsx [hunk 6]
```diff
@@ -1,11 +1,10 @@
 import isPropValid from '@emotion/is-prop-valid'
```
Comment (rafalpawlow) [inline src/components/ProgressDrawer/ProgressDrawer.styles.tsx]: Get rid of `.tsx` extension

File: src/components/Searchbar/Searchbar.styles.tsx [hunk 7]
```diff
@@ -1,13 +1,11 @@
 import isPropValid from '@emotion/is-prop-valid'
```
Comment (rafalpawlow) [inline src/components/Searchbar/Searchbar.styles.tsx]: Get rid of `.tsx` extension

File: src/components/ShortcutIndicator/ShortcutIndicator.styles.tsx [hunk 8]
```diff
@@ -1,6 +1,6 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/components/ShortcutIndicator/ShortcutIndicator.styles.tsx]: Get rid of `.tsx` extension

File: src/components/SignInSteps/TermsStep.styles.tsx [hunk 9]
```diff
@@ -1,8 +1,8 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/components/SignInSteps/TermsStep.styles.tsx]: Get rid of `.tsx` extension

File: src/components/Tabs/Tabs.styles.tsx [hunk 10]
```diff
@@ -1,8 +1,7 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/components/Tabs/Tabs.styles.tsx]: Get rid of `.tsx` extension

File: src/components/_inputs/Checkbox/Checkbox.styles.tsx [hunk 11]
```diff
@@ -1,8 +1,8 @@
 import { css } from '@emotion/react'
```
Comment (rafalpawlow) [inline src/components/_inputs/Checkbox/Checkbox.styles.tsx]: Get rid of `.tsx` extension

File: src/components/_inputs/TitleArea/TitleArea.styles.tsx [hunk 12]
```diff
@@ -1,9 +1,8 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/components/_inputs/TitleArea/TitleArea.styles.tsx]: Get rid of `.tsx` extension

File: src/components/_navigation/TopbarBase/TopbarBase.styles.tsx [hunk 13]
```diff
@@ -1,7 +1,7 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/components/_navigation/TopbarBase/TopbarBase.styles.tsx]: Get rid of `.tsx` extension

File: src/components/_navigation/TopbarStudio/TopbarStudio.styles.tsx [hunk 14]
```diff
@@ -1,10 +1,10 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/components/_navigation/TopbarStudio/TopbarStudio.styles.tsx:1]: Get rid of `.tsx` extension

File: src/components/_navigation/TopbarViewer/TopbarViewer.styles.tsx [hunk 15]
```diff
@@ -1,8 +1,8 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/components/_navigation/TopbarViewer/TopbarViewer.styles.tsx]: Get rid of `.tsx` extension

File: src/components/_video/VideoPlayer/VideoPlayer.styles.tsx [hunk 16]
```diff
@@ -1,13 +1,12 @@
 import { css } from '@emotion/react'
```
Comment (rafalpawlow) [inline src/components/_video/VideoPlayer/VideoPlayer.styles.tsx]: Get rid of `.tsx` extension

File: src/components/_video/VideoTileBase/VideoTileBase.styles.tsx [hunk 17]
```diff
@@ -2,12 +2,11 @@ import { css } from '@emotion/react'
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/components/_video/VideoTileBase/VideoTileBase.styles.tsx]: Get rid of `.tsx` extension

File: src/views/studio/CreateEditChannelView/CreateEditChannelView.style.tsx [hunk 18]
```diff
@@ -1,9 +1,9 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/views/studio/CreateEditChannelView/CreateEditChannelView.style.tsx:1]: Get rid of `.tsx` extension, also let's rename it to `CreateEditChannelView.styles.ts`

File: src/views/studio/CreateMemberView/CreateMemberView.style.ts [hunk 19]
```diff
@@ -1,11 +1,11 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/views/studio/CreateMemberView/CreateMemberView.style.ts:1]: Let's rename it to `CreateMemberView.styles.ts`

File: src/views/studio/MyUploadsView/MyUploadsView.style.tsx [hunk 20]
```diff
@@ -1,8 +1,8 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/views/studio/MyUploadsView/MyUploadsView.style.tsx:1]: Get rid of `.tsx` extension, also let's rename it to `MyUploadsView.styles.ts`

File: src/views/studio/MyUploadsView/UploadStatus/UploadStatus.style.tsx [hunk 21]
```diff
@@ -1,9 +1,9 @@
 import { css } from '@emotion/react'
```
Comment (rafalpawlow) [inline src/views/studio/MyUploadsView/UploadStatus/UploadStatus.style.tsx:1]: Get rid of `.tsx` extension, also let's rename it to `UploadStatus.styles.ts`

File: src/views/studio/MyUploadsView/UploadStatusGroup/UploadStatusGroup.style.tsx [hunk 22]
```diff
@@ -1,9 +1,9 @@
 import { css } from '@emotion/react'
```
Comment (rafalpawlow) [inline src/views/studio/MyUploadsView/UploadStatusGroup/UploadStatusGroup.style.tsx:1]: Get rid of `.tsx` extension, also let's rename it to `UploadStatusGroup.styles.ts`

File: src/views/studio/MyVideosView/NewVideoTile/NewVideoTile.style.ts [hunk 23]
```diff
@@ -1,10 +1,10 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/views/studio/MyVideosView/NewVideoTile/NewVideoTile.style.ts:1]: Let's rename it to `NewVideoTile.styles.ts`

File: src/views/studio/SignInJoinView/SignInMainView/SignInMainView.style.ts [hunk 24]
```diff
@@ -2,10 +2,10 @@ import styled from '@emotion/styled'
 import { Link } from 'react-router-dom'
```
Comment (rafalpawlow) [inline src/views/studio/SignInJoinView/SignInMainView/SignInMainView.style.ts:2]: Let's rename it to `SignInMainView.styles.ts`

**Review by drillprop — COMMENTED** [review]
- agree with Klaudiusz regarding buttons
- I think, we could create `_loaders` directory, where we could keep SkeletonLoader, Loader and Spinner. Not sure if everyone agrees with me.
- DatePicker should be in inputs
- Don't like `_content` directory, because I'm not really sure what we should put there. For example - why didn't you put there for instance `PromisingChannels` or `OfficialJoystreamUpdate`? These two components look very similar to those inside `_content` directory. 
- There are still some of components that are very specific to some views, and Im wondering where we should put them (for example SigninSteps or StudioEntrypoint). I would consider move them to views maybe, but not sure what others think about.

File: src/views/studio/SignInJoinView/SignInProcessView/SignInProcessView.style.ts [hunk 25]
```diff
@@ -1,10 +1,10 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/views/studio/SignInJoinView/SignInProcessView/SignInProcessView.style.ts:1]: Let's rename it to `SignInProcessView.styles.ts`

File: src/views/studio/SignInView/SignInView.style.ts [hunk 26]
```diff
@@ -1,10 +1,10 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/views/studio/SignInView/SignInView.style.ts:1]: Let's rename it to `SignInView.styles.ts`

File: src/views/studio/VideoWorkspace/VideoWorkspace.style.ts [hunk 27]
```diff
@@ -1,7 +1,7 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/views/studio/VideoWorkspace/VideoWorkspace.style.ts:1]: Let's rename it to `VideoWorkspace.styles.ts`

File: src/views/studio/VideoWorkspace/VideoWorkspaceTabsBar/VideoWorkspaceTabsBar.style.ts [hunk 28]
```diff
@@ -1,7 +1,7 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/views/studio/VideoWorkspace/VideoWorkspaceTabsBar/VideoWorkspaceTabsBar.style.ts:1]: Let's rename it to `VideoWorkspaceTabsBar.styles.ts`

File: src/views/viewer/ChannelView/ChannelAbout.style.tsx [hunk 29]
```diff
@@ -1,10 +1,10 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/views/viewer/ChannelView/ChannelAbout.style.tsx:1]: Get rid of `.tsx` extension, also let's rename it to `ChannelAbout.styles.ts`

File: src/views/viewer/ChannelView/ChannelView.style.tsx [hunk 30]
```diff
@@ -2,15 +2,15 @@ import { css } from '@emotion/react'
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/views/viewer/ChannelView/ChannelView.style.tsx:2]: Get rid of `.tsx` extension, also let's rename it to `ChannelView.styles.ts`

File: src/views/viewer/SearchView/SearchResults/SearchResults.style.ts [hunk 31]
```diff
@@ -1,8 +1,8 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/views/viewer/SearchView/SearchResults/SearchResults.style.ts:1]: Let's rename it to `SearchResults.styles.ts`

File: src/views/viewer/VideoView/VideoView.style.tsx [hunk 32]
```diff
@@ -1,10 +1,10 @@
 import styled from '@emotion/styled'
```
Comment (rafalpawlow) [inline src/views/viewer/VideoView/VideoView.style.tsx:1]: Get rid of `.tsx` extension, also let's rename it to `VideoView.styles.ts`

**Review by rafalpawlow — CHANGES_REQUESTED** [review]

**Comment by eldiegod** [comment]
- Buttons in to _buttons folder
- _loaders folder
- TSX to ts style files and style to styles

Not moving components into views yet I think is better to create another ticket to keep scope somewhat focused

Also not clear what to do with the _content folder

**Comment by drillprop** [comment]
> Also not clear what to do with the _content folder

I'd just remove it, because it's unclear to me what we should keep here, but maybe someone have better idea.

**Comment by kdembler** [comment]
Personally, I'd keep it and add the other components you mentioned to it. Maybe we could also move Gallery and other associated components in there?

**Review by kdembler — APPROVED** [review]

**Review by rafalpawlow — APPROVED** [review]

**Pull request merged**